From 3f2de5ab8244f3d8e2663761d77b681825910919 Mon Sep 17 00:00:00 2001
From: "A.I" <ailis@paw.zone>
Date: Mon, 8 May 2017 08:35:29 -0700
Subject: [PATCH] docker-engine: use v1.10.3

It has been moved to https://github.com/moby/moby,
therefor the hash has been changed.
---
 package/docker-engine/Config.in          |  4 ----
 package/docker-engine/docker-engine.hash |  2 +-
 package/docker-engine/docker-engine.mk   | 26 ++++++++++++--------------
 3 files changed, 13 insertions(+), 19 deletions(-)

diff --git a/package/docker-engine/Config.in b/package/docker-engine/Config.in
index 05670a716e..2ffd3f65fc 100644
--- a/package/docker-engine/Config.in
+++ b/package/docker-engine/Config.in
@@ -14,10 +14,6 @@ if BR2_PACKAGE_DOCKER_ENGINE
 config BR2_PACKAGE_DOCKER_ENGINE_DAEMON
 	bool "docker daemon"
 	default y
-	depends on BR2_USE_MMU # docker-containerd
-	select BR2_PACKAGE_DOCKER_CONTAINERD # runtime dependency
-	select BR2_PACKAGE_IPTABLES # runtime dependency
-	select BR2_PACKAGE_SQLITE # runtime dependency
 	help
 	  Build the Docker system daemon.
 	  If not selected, will build client only.
diff --git a/package/docker-engine/docker-engine.hash b/package/docker-engine/docker-engine.hash
index 6c0ed338a5..515f697e86 100644
--- a/package/docker-engine/docker-engine.hash
+++ b/package/docker-engine/docker-engine.hash
@@ -1,2 +1,2 @@
 # Locally calculated
-sha256	4716df117d867b82ddab2e82395cd40aa3d0925a689eedcec8919729e4c9f121	docker-engine-v17.05.0-ce.tar.gz
+sha256	668a571a2406024423f11b4cd150f97b4d7bc3ee39407a0ac7df6e5db1437385	docker-engine-1.10.3.tar.gz
diff --git a/package/docker-engine/docker-engine.mk b/package/docker-engine/docker-engine.mk
index 8928f072e7..d460235871 100644
--- a/package/docker-engine/docker-engine.mk
+++ b/package/docker-engine/docker-engine.mk
@@ -4,16 +4,16 @@
 #
 ################################################################################
 
-DOCKER_ENGINE_VERSION = v17.05.0-ce
-DOCKER_ENGINE_COMMIT = 89658bed64c2a8fe05a978e5b87dbec409d57a0f
-DOCKER_ENGINE_SITE = $(call github,docker,docker,$(DOCKER_ENGINE_VERSION))
+DOCKER_ENGINE_VERSION = 1.10.3
+DOCKER_ENGINE_GITCOMMIT = 20f81dd
+DOCKER_ENGINE_SITE = $(call github,moby,moby,v$(DOCKER_ENGINE_VERSION))
 
 DOCKER_ENGINE_LICENSE = Apache-2.0
 DOCKER_ENGINE_LICENSE_FILES = LICENSE
 
 DOCKER_ENGINE_DEPENDENCIES = host-go host-pkgconf
 
-DOCKER_ENGINE_GOPATH = "$(@D)/gopath"
+DOCKER_ENGINE_GOPATH = "$(@D)/vendor"
 DOCKER_ENGINE_MAKE_ENV = $(HOST_GO_TARGET_ENV) \
 	CGO_ENABLED=1 \
 	CGO_NO_EMULATION=1 \
@@ -22,8 +22,8 @@ DOCKER_ENGINE_MAKE_ENV = $(HOST_GO_TARGET_ENV) \
 	PKG_CONFIG="$(PKG_CONFIG_HOST_BINARY)" \
 	$(TARGET_MAKE_ENV)
 
-DOCKER_ENGINE_GLDFLAGS = \
-	-X main.GitCommit=$(DOCKER_ENGINE_VERSION) \
+DOCKER_ENGINE_GLDFLAGS = -w -s \
+	-X main.GitCommit=$(DOCKER_ENGINE_GITCOMMIT) \
 	-X main.Version=$(DOCKER_ENGINE_VERSION)
 
 ifeq ($(BR2_STATIC_LIBS),y)
@@ -34,7 +34,8 @@ DOCKER_ENGINE_GLDFLAGS_DOCKER += -extldflags '-static'
 endif
 endif
 
-DOCKER_ENGINE_BUILD_TAGS = cgo exclude_graphdriver_zfs autogen
+DOCKER_ENGINE_BUILD_TAGS = cgo exclude_graphdriver_zfs exclude_graphdriver_aufs autogen \
+	sqlite_omit_load_extension
 DOCKER_ENGINE_BUILD_TARGETS = docker
 
 ifeq ($(BR2_PACKAGE_LIBSECCOMP),y)
@@ -49,7 +50,6 @@ endif
 
 ifeq ($(BR2_PACKAGE_DOCKER_ENGINE_DAEMON),y)
 DOCKER_ENGINE_BUILD_TAGS += daemon
-DOCKER_ENGINE_BUILD_TARGETS += dockerd
 endif
 
 ifeq ($(BR2_PACKAGE_DOCKER_ENGINE_EXPERIMENTAL),y)
@@ -75,12 +75,11 @@ DOCKER_ENGINE_BUILD_TAGS += exclude_graphdriver_vfs
 endif
 
 define DOCKER_ENGINE_CONFIGURE_CMDS
-	mkdir -p $(DOCKER_ENGINE_GOPATH)/src/github.com/docker
 	ln -fs $(@D) $(DOCKER_ENGINE_GOPATH)/src/github.com/docker/docker
 	cd $(@D) && \
-		GITCOMMIT="$$(echo $(DOCKER_ENGINE_COMMIT) | head -c7)" \
+		GITCOMMIT="$(DOCKER_ENGINE_GITCOMMIT)" \
 		BUILDTIME="$$(date)" \
-		VERSION="$(patsubst v%,%,$(DOCKER_ENGINE_VERSION))" \
+		VERSION="$(DOCKER_ENGINE_VERSION)" \
 		PKG_CONFIG="$(PKG_CONFIG_HOST_BINARY)" $(TARGET_MAKE_ENV) \
 		bash ./hack/make/.go-autogen
 endef
@@ -105,13 +104,12 @@ endif
 
 define DOCKER_ENGINE_BUILD_CMDS
 	$(foreach target,$(DOCKER_ENGINE_BUILD_TARGETS), \
-		cd $(@D)/gopath/src/github.com/docker/docker; \
-		$(DOCKER_ENGINE_MAKE_ENV) \
+		cd $(@D); $(DOCKER_ENGINE_MAKE_ENV) \
 		$(HOST_DIR)/bin/go build -v \
 			-o $(@D)/bin/$(target) \
 			-tags "$(DOCKER_ENGINE_BUILD_TAGS)" \
 			-ldflags "$(DOCKER_ENGINE_GLDFLAGS) $(DOCKER_ENGINE_GLDFLAGS_$(call UPPERCASE,$(target)))" \
-			github.com/docker/docker/cmd/$(target)
+			./$(target)
 	)
 endef
 
-- 
2.14.1

